home/export_ordered_sets.py

```diff
@@ -0,0 +1,172 @@
+        The field names, used to write the header rows in exports
+        """
+        return [f.name for f in fields(cls)]
+
+    def to_dict(self) -> dict[str, str]:
+        return asdict(self)
+
+    def to_tuple(self) -> tuple[str]:
+        return astuple(self)
+
+
+class OrderedSetExporter:
+    rows: list[ExportRow]
+
+    def __init__(self, queryset: PageQuerySet):
+        self.queryset = queryset
+        self.rows = []
+
+    def perform_export(self) -> Iterable[ExportRow]:
+        for item in self.queryset:
+            yield ExportRow(
+                name=item.name,
+                profile_field=item.profile_field,
+                page_slugs=item.page_slugs,
+                page=item.page,
+                unit=item.unit,
+                before_or_after=str(item.before_or_after),
+                contact_field=item.contact_field,
+            )
+
+
+def serialize_list(items: Iterable[str]) -> str:
+    """
+    Uses CSV formatting to seralize a list of strings, handling escaping
+    """
+    output = io.StringIO()
+    writer = csv.writer(output)
+    writer.writerow(items)
+    return output.getvalue().rstrip("\r\n")
+
+
+class OrderedSetsExportWriter:
+    rows: Iterable[ExportRow]
+
+    def __init__(self, rows: Iterable[ExportRow]):
+        self.rows = rows
+
+    def write_xlsx(self, response: HttpResponse) -> None:
+        workbook = Workbook()
+        worksheet: Worksheet = workbook.active  # type: ignore  # This is not a write or read only workbook
+        worksheet.append(ExportRow.headings())
+        for row in self.rows:
+            worksheet.append(row.to_tuple())
+        _set_xlsx_styles(workbook, worksheet)
+        workbook.save(response)
+
+    def write_csv(self, response: HttpResponse) -> None:
+        writer = csv.DictWriter(f=response, fieldnames=ExportRow.headings())
+        writer.writeheader()
+
+        for row in self.rows:
+            writer.writerow(row.to_dict())
+
+
+def _set_xlsx_styles(wb: Workbook, sheet: Worksheet) -> None:
+    """
+    Sets the style for the workbook adding any formatting that will make the sheet more
+    aesthetically pleasing
+    """
+    # Adjustment is because the size in openxlsx and google sheets are not equivalent
+    adjustment = 7
+
+    # Set columns based on best size
+    column_widths_in_pts = {
+        "title": 110,
+        "question_type": 110,
+        "tags": 110,
+        "slug": 110,
+        "version": 90,
+        "locale": 50,
+        "high_result_page": 110,
+        "high_inflection": 110,
+        "medium_result_page": 120,
+        "medium_inflection": 110,
+        "low_result_page": 110,
+        "skip_threshold": 110,
+        "skip_high_result_page": 110,
+        "generic_error": 370,
+        "question": 370,
+        "explainer": 370,
+        "error": 370,
+        "min": 110,
+        "max": 110,
+        "answers": 370,
+        "scores": 110,
+        "answer_semantic_ids": 110,
+        "question_semantic_id": 110,
```

Comment:
Most of these columns don't appear in ordered content sets. The ones that aren't in ordered content sets should be removed, and the missing ones added.